samples: DFU over Bluetooth SMP rework (#117)
nrfconnect/ncs-zigbee: docs/includes/matter_extension_bt_dfu.txt

Device firmware upgrade over Bluetooth SMP

> --------------------------------------------

Reviewer: ```suggestion
------------------------------------------
```

The Matter extension builds include Matter over-the-air (OTA) update support through MCUboot by default.

> Device Firmware Upgrade (DFU) over Bluetooth LE using the Simple Management Protocol (SMP) is also enabled by default in the :file:`prj_matter_fota.conf` configuration selected by ``FILE_SUFFIX=matter_fota``.

Reviewer: What do you think about providing a link to what SMP is (Zephyr docs)?

Matter OTA and SMP DFU are independent transports.

> SMP is not part of the Matter specification.

Reviewer: I would add one more sentence that Matter per spec requires supporting at least one method of performing software update over the air, and here we have two of then enabled just to showcase how to use them.

Device firmware upgrade over Bluetooth SMP
------------------------------------------

The Matter extension builds include Matter over-the-air (OTA) update support through MCUboot by default.

> Device Firmware Upgrade (DFU) over Bluetooth LE using the `Simple Management Protocol`_ (SMP) is also enabled by default in the :file:`prj_matter_fota.conf` configuration selected by ``FILE_SUFFIX=matter_fota``.

Reviewer 2: ```suggestion
You can update your device's firmware over the air using two methods, both enabled by default in the Matter extension builds:

* Matter OTA – Updates your device through MCUboot, satisfying the Matter specification's requirement to support at least one over-the-air software update method.
* Device Firmware Upgrade (DFU) over Bluetooth LE – Updates your device using the `Simple Management Protocol`_ (SMP). 
  To enable this method, select the :file:`prj_matter_fota.conf` configuration by setting the  ``FILE_SUFFIX=matter_fota`` option.
 
```

The Matter specification requires supporting at least one over-the-air software update method; Matter OTA fulfills this requirement.
SMP DFU is an independent transport and is not part of the Matter specification.
Both transports are enabled in this configuration to showcase how to use them.

Use `mcumgr`_ or `nRF Connect Device Manager`_ to perform the update.

> See |sample matter bt dfu testing ref| for step-by-step testing instructions.

Reviewer 2: ```suggestion
For step-by-step instructions on testing the update process, see |sample matter bt dfu testing ref|.
```
